### PR TITLE
feat: highlight fragement headings when clicked

### LINKED ISSUE #303: Highlight fragment headings when clicked
#### Reason/Context
When visiting https://json-schema.org/learn/glossary#meta-schema or some other link to a fragment on the site, it would be nice if some highlighting indicated the active fragment / heading.

#302 is related inasmuch as when you visit a fragment link where the heading is one of the *bottom* ones of the page, they may *not* be at the top of the viewport, so highlighting is nice to make it clear where someone should look

#### Description
Highlight the active heading in some highlight color when the heading is clicked, or when visited via a fragment.

**Do you think this work might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

Yes/No
No

### PR DESCRIPTION
**Pull request by utnim2**
Fixes #303 

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No
[JUSTIFICATION]

### CODE REVIEW

**Comment by Min2who** [comment]
**DONT MERGE THIS**

As would like to refractor some code and want to close #303  in this PR only

**I dont know why i am getting hydration errors**

**Comment by benjagm** [comment]
Thanks for this PR @Min2who . Can you please describe what are you fixing? I am not sure what is not working as expected and the video has no explanation or voice so an additional explanation will be appreciated.

Thanks!!

**Review by benjagm — CHANGES_REQUESTED** [review]
Thanks a lot for your PR!!

I think the expected behaviour is to highlight just the title of the section your are visiting, no the one you are scrolling over, so the highlight will only happen when you actually click the section header, no when you mouse over what seems to be your implementation.

In addition I am getting some errors when testing your code (see attached):
Screenshot 2024-02-15 at 11 07 33

Finally, I thinks the blue background is not ideal for highlight. I'll say will be better something more subtle. Maybe underline or adding an icon or emoji ...

**Comment by Min2who** [comment]
> Thanks a lot for your PR!!
> 
> I think the expected behaviour is to highlight just the title of the section your are visiting, no the one you are scrolling over, so the highlight will only happen when you actually click the section header, no when you mouse over what seems to be your implementation.
> 
> In addition I am getting some errors when testing your code (see attached): Screenshot 2024-02-15 at 11 07 33
> 
> Finally, I thinks the blue background is not ideal for highlight. I'll say will be better something more subtle. Maybe underline or adding an icon or emoji ...

I will be working on it as soon as me exam get over. 🙇‍♂️

**Comment by benjagm** [comment]
@Min2who remember you have this PR with some small changes to be added!

**Comment by Min2who** [comment]
> In addition I am getting some errors when testing your code (see attached): Screenshot 2024-02-15 at 11 07 33
> 
@benjagm after many trials of solving this error , I found that the error to be there from the start and occur even when i remove my changes it is still present 

Should i proceed with the PR of changes or what shall i do a comment will be appreciated

**Comment by benjagm** [comment]
> @benjagm after many trials of solving this error , I found that the error to be there from the start and occur even when i remove my changes it is still present
> 
> Should i proceed with the PR of changes or what shall i do a comment will be appreciated

No sure what you mean here. Is your code fixing the bug? If yes push the code.

**Comment by benjagm** [comment]
@DarhkVoyd can I ask you help with this PR? @Min2who seems to have issues with the code. Maybe you can take a look.
Thanks a lot for your support.

**Comment by DarhkVoyd** [comment]
@benjagm Absolutely, I'd be happy to help out! I'll take a look and see if I can provide any assistance or guidance. Thanks for reaching out.

**Comment by DarhkVoyd** [comment]
@benjagm The hydration issue is not related to this PR, the website already has multiple hydration issues originating from different sections in `Layout` and `Sidebar` components due to client side changes on server components. The production deployed website too displays minified errors for this in dev tools. Take a look: https://json-schema.org/learn/getting-started-step-by-step#overview

**Comment by DarhkVoyd** [comment]
@Min2who you should probably proceed with the PR and make the changes requested in the last review.

**Comment by Min2who** [comment]
https://www.loom.com/share/b9310a18a5744a26a125047b52c9af44?sid=6917409a-8c2a-4b11-81cb-1750d19e3f82

Done with the changes mentioned @benjagm

**Comment by benjagm** [comment]
@Min2who You just submitted a new PR without addressing the [comment](https://github.com/json-schema-org/website/pull/304#pullrequestreview-1882362056) I did in my first review. I have been waring about being careful about pushing new PRs without addressing the review comments and without testing properly. This is time confusing and frustrating for others reviewing your code. 

At this point I am going to cancel this PR and leave the issue available for other contributor.
All reach out to you in a different message with more instructions.

**Comment by benjagm** [comment]
https://github.com/json-schema-org/website/pull/304#issuecomment-1968891764

**Comment by Min2who** [comment]
> @Min2who You just submitted a new PR without addressing the [comment](https://github.com/json-schema-org/website/pull/304#pullrequestreview-1882362056) I did in my first review. I have been waring about being careful about pushing new PRs without addressing the review comments and without testing properly. This is time confusing and frustrating for others reviewing your code.
> 

where is that new PR. If you find it please let me know @benjagm

**Comment by utnim2** [comment]
> I think the expected behaviour is to highlight just the title of the section your are visiting, no the one you are scrolling over, so the highlight will only happen when you actually click the section header, no when you mouse over what seems to be your implementation.
> 
> Finally, I thinks the blue background is not ideal for highlight. I'll say will be better something more subtle. Maybe underline or adding an icon or emoji ...

Your requested changes were addresed !
https://www.loom.com/share/b9310a18a5744a26a125047b52c9af44?sid=6917409a-8c2a-4b11-81cb-1750d19e3f82

**Comment by VivekJaiswal18** [comment]
@benjagm can I be assigned this issue. Thank you !

**Comment by DarhkVoyd** [comment]
@benjagm Incase, we are unsure about the styles. How about something like [this](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#baseProtocol).

**Comment by benjagm** [comment]
@DarhkVoyd and @utnim2 we'd love you to collaborate to complete this issue. DV can you please do the PR review and suggest changes to Mintu and as soon as this is working and approved as me for review? 

I need some extra hands to help me with the amount of PRs I have to review.

**Comment by DarhkVoyd** [comment]
@benjagm Sure, I would love to help out. Any suggestions or advice on current status of the PR?

**Comment by benjagm** [comment]
> Any suggestions or advice on current status of the PR? I'll take it from here and reach out once it's ready for your review.

You are doing the PR review: Just clone the repo and test it to is if works. If works review the code to see if all looks good... and if everything is ok may suggest some fine-tuning of the style.

**Comment by utnim2** [comment]
https://www.loom.com/share/87723b2cd1d2439dbc73495335c2fe3c?sid=b4fdac1a-4e6b-47b5-8c63-9d6adf31c0fc

The changes that i have made 

Cc @benjagm @DarhkVoyd

File: components/Headlines.tsx
Comment (DarhkVoyd) [inline components/Headlines.tsx:30]: Here, the previous implementation states 
> // recalculation necessary because of scope issue

This PR changes it, why is it no longer necessary

**Review by DarhkVoyd — CHANGES_REQUESTED** [review]
Nice work with this one, the code works nicely. I added some comments please check them. Also,
- I believe your editor's formatter has auto-formatted code on save. Let's keep our PRs focused on the changes that we are intentionally making to avoid adding unnecessary noise.
- I somewhat agree with your current highlighting approach but I believe it's not enough let's make it bit more visible. What do you think? How about a subtle grey background?

**Comment by DarhkVoyd** [comment]
I agree with @benjagm's suggestion, however, I'm concerned readers might still miss it.  Could we experiment with a slightly lighter background color like a subtle shade of grey? Or perhaps a combination of a light background and underlined text? What do you think?

**Comment by divyaxdv** [comment]
> I agree with @benjagm's suggestion, and I even missed the highlight initially. I'm concerned readers might have the same problem. Could we experiment with a slightly lighter background color like a subtle shade of grey? Or perhaps a combination of a light background and underlined text? What do you think?

I think the changes does its part pretty well, but I agree a little can be improvised like a background to make the fragment more distinguished.

**Comment by benjagm** [comment]
@utnim2 can you please rebase and fix the merge conflicts please? I'll make a another review.

**Comment by benjagm** [comment]
We are closing this PR as we didn't receive response.

**Comment by utnim2** [comment]
Sorry for that @benjagm , I will help to review the PR

**Pull request closed without merging**
